js/index.js

```diff
@@ -0,0 +1,210 @@
+'use strict'
+
+const blinding = (iter, on = null) => {
+    const action = on == null ?
+        "toggle" :
+        on ?
+            "add" :
+            "remove"
+
+    for (const item of iter) {
+        item.classList[action]('blind');
+    }
+};
+
+const toHtml = source => {
+    let $div = document.createElement("div");
+    $div.innerHTML = source;
+    return $div.firstElementChild;
+}
+
+class TodoApp {
+    constructor() {
+        this.$todoInput = document.getElementById("new-todo-title");
+        this.$todoList = document.getElementById("todo-list");
+        this.$todoFooter = document.querySelector(".count-container");
+        this.previous = "";
+        this.build();
+        this.eventRegist();
+    }
+
+    getTodoEntity(idx) {
+        let $todoItem = this.$todoList.querySelector(`[data-idx='${idx}']`);
+        return {
+            idx:Number(idx),
+            value:$todoItem.querySelector('.label').textContent,
+            check:$todoItem.querySelector('.toggle').checked
+        }
+    }
+
+    getLastIndex() {
+        let todoItems = lsg.get('todoItems');
+        let max = 0;
+        if (!todoItems || todoItems.length === 0) return max;
+
+        for (const item of todoItems) {
+            if (max < item.idx) max = item.idx;
+        }
+        return max;
+    }
+
+    getNextIndex() {
+        return this.getLastIndex() + 1;
+    }
+
+    save = (entity) => {
+        let todoItems = lsg.get("todoItems");
+        if (!todoItems) todoItems = [];
+        let findItem = todoItems.find(a => a.idx === entity.idx);
+        if (findItem) {
+            findItem.value = entity.value;
+            findItem.check = entity.check;
+        } else {
+            todoItems.push(entity);
+        }
+        lsg.set('todoItems', todoItems);
+    }
+    delete = (idx) => {
+        if(whatType(idx)==="string")idx = Number(idx);
+        let todoItems = lsg.get("todoItems");
+        let newTodoItems = [];
+        if (todoItems.length > 0) {
+            for (const todoItem of todoItems) {
+                if (todoItem.idx !== idx) {
+                    newTodoItems.push(todoItem);
+                }
+            }
+        }
+        lsg.set('todoItems', newTodoItems);
```

Comment:
```suggestion
        const todoItems = lsg.get("todoItems");
        const key = todoItems.findIndex(v => v.idx === idx);
        todoItems.splice(key, 1);
        lsg.set('todoItems', todoItems);
```
splice와 findIndex를 이용하여 이렇게 표현할 수 있습니다!